### PR TITLE
Don't do direct casts to int to prevent dart2wasm breakages.

### PR DESCRIPTION
**Pull request by eyebrowsoffire**
This fixes https://github.com/dart-lang/test/issues/2006

### CODE REVIEW

**Comment by natebosch** [comment]
Are there tests in this repo that will fail without this change so that we don't regress?

**Review by jakemac53 — APPROVED** [review]

**Comment by jakemac53** [comment]
> Are there tests in this repo that will fail without this change so that we don't regress?

It looks like if we made a dart2wasm test that uses a timeout we could get that

**Comment by eyebrowsoffire** [comment]
I'll write a test

**Comment by jakemac53** [comment]
See https://github.com/dart-lang/test/tree/master/integration_tests/wasm which is probably the easiest place to put this test for now

File: integration_tests/wasm/mono_pkg.yaml [hunk 1]
```diff
@@ -18,6 +18,6 @@ stages:
       - which google-chrome-beta
       os:
         - linux
-    - test:
+    - test: --timeout=60s --retry=2
```
Comment (jakemac53) [inline integration_tests/wasm/mono_pkg.yaml:21]:
```suggestion
    # The config here is a regression test for https://github.com/dart-lang/test/issues/2006
    - test: --timeout=60s --retry=2
```

Comment (jakemac53) [inline integration_tests/wasm/mono_pkg.yaml:21]: Note that you actually can't configure the number of retries here (its just a flag to enable or disable retries). The number of retries can be configured in dart_test.yml though.

File: integration_tests/wasm/test/hello_world_test.dart [hunk 2]
```diff
@@ -1,7 +1,9 @@
 // Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
+
 @TestOn('wasm')
+@Retry(2)
```
Comment (jakemac53) [inline integration_tests/wasm/test/hello_world_test.dart:6]:
```suggestion
# This retry is a regression test for https://github.com/dart-lang/test/issues/2006
@Retry(2)
```

**Review by jakemac53 — APPROVED** [review]

**Comment by jakemac53** [comment]
@eyebrowsoffire did you need a publish of this soon? @natebosch thoughts on when we should publish?

**Comment by eyebrowsoffire** [comment]
We have a workaround in Flutter for now, so there's no huge rush on a publish, but let me know there is a new version and we'll bump to it.

**Pull request merged**